scripts/Molecule-example.jl

```diff
@@ -0,0 +1,68 @@
+# # Example 6: Molecular Hamiltonian Calculation
+
+# This is an example calculation of the water molecule (H₂O).
+
+# A runnable script for this example is located
+# [here](https://github.com/RimuQMC/Rimu.jl/blob/develop/scripts/Molecule-example.jl).
+# Run it with `julia Molecule-example.jl`.
+
+# First, we load the reqired packages. `Rimu` for FCIQMC calculation.
+using Rimu
+
+# ## Setting up the model
+
+# We specify the path to the FCIDUMP file describing the H₂O molecule.
+fcidump = joinpath(pkgdir(Rimu), "test/examples/h2o.FCIDUMP");
+
+# Next we construct the Hamiltonian by the constructor [`MolecularHamiltonian`](@ref). 
+# The Hartree-Fock ground state is generated automatically at the same time. 
+h = MolecularHamiltonian(fcidump)
+a = starting_address(h);
+
+# ## Running the calculation
+
+# ### Exact Diagonalization calculation
+# We first define the problem via [`ExactDiagonalizationProblem`](@ref) interface.
+# Since H₂O has a rather large Hilbert space, a iterative solver is recomended. We
+# use [`KrylovKitSolver`](@ref). It requires [KrylovKit.jl](https://github.com/Jutho/KrylovKit.jl).
+# To install it, in Julia REPL type ] to enter the Pkg REPL mode and run:
+# ```julia-repl
+# pkg> add KrylovKit
+# ``` 
+# Or via the `Pkg` package
+# ```julia-repl
+# julia> import Pkg; Pkg.add("KrylovKit.jl")
+# ``` 
+using KrylovKit
+
+p = ExactDiagonalizationProblem(h, algorithm=KrylovKitSolver(true))
+
+# Then it can be solved by executing
+# ```julia
+# s = solve(p)
+# ```
```

Comment:
Maybe we should warn here that this may require larger computational resources.

I'm actually not sure what exactly happens and why this seems to stall. `solve(p; full_basis=true)` seems a good idea, as initialising the `LinearOperator` is done much faster (12s on my laptop), but the matrix-vector multiplications seem to stall. We can deal with this later, though, as I don't think it is an issue related to this PR.